### PR TITLE
Remove portion of test from matrix suite and add to host tests.

### PR DESCRIPTION
**Pull request by lukemelia**
The Matrix suite had an AI Assistant test that mutates realm state, which is not desired in that suite. This commit removes that portion of the matrix suite test and reproduces the test in the host suite. Making it work required making the MockMatrixService a bit more faithful to the actual one.

### CODE REVIEW

File: packages/host/tests/acceptance/ai-assistant-test.gts [hunk 1]
```diff
@@ -0,0 +1,306 @@
...
+    // );
+    // cardFragmentEvents = events.filter(
+    //   (e) =>
+    //     e.type === 'm.room.message' &&
+    //     e.content.msgtype === 'org.boxel.cardFragment' &&
+    //     !e.content.data.nextFragment,
+    // );
+    // expect(messageEvents.length).toEqual(4);
+    // expect(cardFragmentEvents.length).toEqual(3);
+  });
```
Comment (backspace) [inline packages/host/tests/acceptance/ai-assistant-test.gts:289]: Is this meant to be commented out? It looks like it wasn’t in the original test.

**Review by habdelra — APPROVED** [review]
looks good (aside from buck's comment)

**Review by jurgenwerk — APPROVED** [review]

**Review by burieberry — APPROVED** [review]
Looks good, but same question as @backspace regarding the commented out section

File: packages/host/tests/acceptance/ai-assistant-test.gts (hunk 1 above)
Comment (lukemelia) [reply packages/host/tests/acceptance/ai-assistant-test.gts:289]: Whoops, those checks were done in the matrix suite version, but I think it is better for this acceptance test to not to assert on the inner workings and instead only validate what the user experience is.

**Pull request merged**